### ISSUE TITLE
Executables cookbook, imcopy, smem, and speed missing from cmake configuration

### ISSUE THREAD

**Issue by drbixx** [HEASARC/cfitsio#33]
The executables `cookbook`, `imcopy`, `smem`, and `speed` are missing from `CMakeList.txt`. As a consequence, they have started missing from the cfitsio conda package because the cfitsio feedstock has switched to a cmake build.

Could the missing executables be added to `CMakeList.txt`?

**Comment by bryanirby** [comment]
Hi @drbixx - Yes, we can add those.  The cookbook executable can be built using -DTESTS=ON, but historically several of the utilities (including the other ones you list) were only built optionally.  We added them to the default libtool build when we updated that configuration recently, but will synchronize the cmake build accordingly.

**Comment by drbixx** [comment]
Great! Thanks @bryanirby

**Comment by rmathar** [comment]
The disadvantage of compiling these is that someone calling `make install` (on the
automake side) pushes these into `/usr/bin` where the names are too general.
(There is for example also a cookbook in `CCfits` which would land at the same
place as the cookbook of cfitsio. How would a package manager know which to
remove in an uninstall?) So the programs should rather be `cfits_cookbook`,
`cfits_speed` or something of that kind to avoid name clashes with other programs.
For these reasons I excluded `cookbook`, `speed` and `smem` in the builds of 
https://github.com/HEASARC/cfitsio/pull/53/files

Apparently gentoo is kicking out `smem` during the build (my naive interpretation of `RDEPEND` in the script in https://gitweb.gentoo.org/repo/gentoo.git/tree/sci-libs/cfitsio/cfitsio-4.6.2.ebuild )

**Comment by bryanirby** [comment]
Thanks @rmathar - historically those tools were an optional install ("make utils") and have been inconsistently handled in the autotools and CMake configurations.  When I did away with the 'utils' target in a recent autotools update I overlooked the fact that the names are too general.  Going forward I'm considering restoring 'utils' in autotools to be more consistent with CMake, and taking speed and cookbook (and others?) out of the install, i.e. to have them only available as local executables (does anyone use 'speed'?).  I'm open to more input along these lines if there are preferences.

**Comment by rmathar** [comment]
Yes, I used speed because there are colleagues who want to install a 8k x 8k camera on the LBT table, and there was the interesting question whether the byte-endianess-swapping would slow down pushing these 16-bit images into FITS files. (We managed to store essentially the full 660 MB/s offered in stream mode by the CCD's, played around with +O3, checked effects of replacing the assembly cod in `swapproc.c` by headers fearing it may slow down things for AVX512 processors..).  Of course a "local binary compiled" `speed` is all that is needed.
